ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFBaseTrim.java

```diff
@@ -68,11 +82,24 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
     if (valObject == null) {
       return null;
     }
-    String val = ((Text) converter.convert(valObject)).toString();
+    String val = stringToTrimConverter.convert(valObject).toString();
     if (val == null) {
       return null;
     }
-    result.set(performOp(val.toString()));
+
+    String trimChars = " ";
```

Comment:
there is also some vectorized implementations (see `StringRTrim` for example)

the functionality is enhanced a bit here - those other implementations should be updated as well (and possibly also covered with tests)

Reply:
I add supporting vectorized two parameter version of trim functions when the first parameter is a column the second is a literal. Example: 
```
create table t1 (col0 string);
select trim(col0, 'xy') from t1 group by col0;
```
In case of trim chars parameter is also column we fall back for non-vectorized version of trim. I guess this use case is not as common as the previous but it can be implemented in a follow-up patch.